### DISCUSSION TITLE
thread 'main' panicked at 'Unable to find a GPU! Make sure you have installed required drivers!'

### DISCUSSION THREAD

**Discussion by tsklxiv** [bevyengine/bevy#2392] [Q&A]
# Information
- Bevy 0.5.0
- Alter Linux (based on Arch Linux)

Hey everyone! I need help.

When I run Bevy with `DefaultPlugins`, like this:

```rust
use bevy::prelude::*;
fn main() {
  App::build().add_plugins(DefaultPlugins).run();
}
```

I got this error:

```
thread 'main' panicked at 'Unable to find a GPU! Make sure you have installed required drivers!', 
/home/firefly/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_wgpu-0.5.0/src/wgpu_renderer.rs:47:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

What should I do?

**Comment by Ixentus** [comment]
What GPU do you have and what drivers have you installed for it?

**Reply by tsklxiv** [reply]
I have an Intel Mesa GPU and I have installed some Vulkan drivers: The Intel Mesa Vulkan driver (`vulkan-intel`), Mesa's Vulkan layer (`vulkan-mesa-layer`) and `vulkan-devel`.

**Reply by Ixentus** [reply]
Can you run any of the wgpu examples? https://github.com/gfx-rs/wgpu

**Comment by thalesmg** [comment]
I have the same problem, running on Arch Linux and Bevy 0.5.0.

I have an older graphics card (NVIDIA Corporation GT218M [GeForce 310M]) that uses the NVIDIA 340xx drivers, and I have `vulkan-intel` installed.

```sh
$ sudo lshw -c display
  *-display                 
       description: VGA compatible controller
       product: GT218M [GeForce 310M]
       vendor: NVIDIA Corporation
       physical id: 0
       bus info: pci@0000:02:00.0
       version: a2
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi pciexpress vga_controller bus_master cap_list rom
       configuration: driver=nvidia latency=0
       resources: irq:28 memory:dc000000-dcffffff memory:e0000000-efffffff memory:de000000-dfffffff ioport:2000(size=128) memory:c0000-dffff
```
I tried to run the `hello` wgpu example per [@Ixentus 's suggestion](https://github.com/bevyengine/bevy/discussions/2392#discussioncomment-929074), and it indeed fails:

```
$ cargo run --example hello
    Finished dev [unoptimized + debuginfo] target(s) in 7.41s
     Running `target/debug/examples/hello`
[2021-10-01T12:15:19Z ERROR wgpu_hal::vulkan::instance] GENERAL [Loader Message (0x0)]
    	setup_loader_term_phys_devs:  Failed to detect any valid GPUs in the current config
[2021-10-01T12:15:19Z ERROR wgpu_hal::vulkan::instance] 	objects: (type: INSTANCE, hndl: 0x55fea887ea00, name: ?)
[2021-10-01T12:15:19Z ERROR wgpu_hal::vulkan::instance] GENERAL [Loader Message (0x0)]
    	setup_loader_tramp_phys_devs:  Failed during dispatch call of 'vkEnumeratePhysicalDevices' to lower layers or loader to get count.
[2021-10-01T12:15:19Z ERROR wgpu_hal::vulkan::instance] 	objects: (type: INSTANCE, hndl: 0x55fea887ea00, name: ?)
[2021-10-01T12:15:19Z ERROR wgpu_hal::vulkan::instance] enumerate_adapters: Initialization of a object has failed
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', wgpu/examples/hello/main.rs:7:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
``` 

Is there some configuration I can tweak? Or is this card too old to ever run Bevy/wgpu?

**Reply by bjorn3** [reply]
According to https://wiki.debian.org/NvidiaGraphicsDrivers the proprietary nvidia 340 series graphics drivers doesn't support Vulkan. Bevy will likely update to wgpu 0.10 once the 2021 edition is released, which has OpenGL support. We are waiting for the 2021 edition as wgpu 0.10 requires the v2 dependency resolver of cargo, which will be the default in the 2021 edition.

**Reply by thalesmg** [reply]
Thanks for the fast answer and for the reference! I guess I'll have to wait and use a newer machine until then. =)
